### PR TITLE
Fix crash when rotating screen in XR mode

### PR DESCRIPTION
**Pull request by bghgary**

### CODE REVIEW

**Comment by bghgary** [comment]
@CedricGuillemet Can you test this with your devices?

**Comment by CedricGuillemet** [comment]
> @CedricGuillemet Can you test this with your devices?

Tested OK on device. No crash when rotating screen.

File: Dependencies/xr/Include/XR.h [hunk 1]
```diff
@@ -333,7 +333,7 @@ namespace xr
             // Move to private when changing to unique_ptr.
             Session(System& system, void* graphicsDevice, std::function<void*()> windowProvider);
 
-            std::unique_ptr<Frame> GetNextFrame(bool& shouldEndSession, bool& shouldRestartSession, std::function<void(void* texturePointer)> deletedTextureCallback = [](void*){});
+            std::unique_ptr<Frame> GetNextFrame(bool& shouldEndSession, bool& shouldRestartSession, std::function<arcana::task<void, std::exception_ptr>(void*)> deletedTextureAsyncCallback = [](void*){ return arcana::task_from_result<std::exception_ptr>(); });
```
Comment (ryantrem) [inline Dependencies/xr/Include/XR.h:336]: Can we default this to `nullptr`?

File: Dependencies/xr/Source/ARCore/XR.cpp [hunk 2]
```diff
@@ -1080,19 +1080,15 @@ namespace xr
             }
         }
 
-        void DestroyDisplayResources(std::function<void(void* texturePointer)> deletedTextureCallback = [](void*){})
+        void DestroyDisplayResources(std::function<arcana::task<void, std::exception_ptr>(void* texturePointer)> deletedTextureAsyncCallback = [](void*){ return arcana::task_from_result<std::exception_ptr>(); })
```
Comment (ryantrem) [inline Dependencies/xr/Source/ARCore/XR.cpp:1083]: Why do we need to specify the default value a second time here? Sorry can't remember.

File: Dependencies/xr/Include/XR.h (hunk 1 above)
Comment (bghgary) [reply Dependencies/xr/Include/XR.h:336]: We can, but I don't like it. The logic is more complicated.

File: Dependencies/xr/Source/ARCore/XR.cpp (hunk 2 above)
Comment (bghgary) [reply Dependencies/xr/Source/ARCore/XR.cpp]: `DestroyDisplayResources` is called without the last parameter in the destructor.

**Review by ryantrem — APPROVED** [review]

**Pull request merged**